### DISCUSSION TITLE
iOS Support?

### DISCUSSION THREAD

**Discussion by derselbst** [FluidSynth/fluidsynth#882] [Ideas]
We have a request to create binaries for iOS as well. Anybody interested? Pls. upvote.

**Comment by Vasa2010** [comment]
Wow !!! This it will be really great !!
I have been coding many years using coreAudio & CoreMidi and che AVFoundantion class for SF2 is not very good in performance !!
Recently I started using JUCE and with some example I did code an SF2 Sampler.
Unfortunately is not possible to use it in iOS Project,
So, your idea will be great!!!
Let me know

**Comment by PaulMcClernan** [comment]
Yes! I've made a FluidSynth wrapper for the LiveCode IDE/ScriptEngine and so far works great on all platforms that LC supports (MacOS, Win, Lin, Android) but there's no binary for iOS. Of course, as Vasa2010 touched on, there is a DLS/SF2 capable Apple Audio Unit Sampler, but it world be great to have a single playback engine that could be deployed on all the platforms.

**Comment by derselbst** [comment]
My idea would be to set up an Azure Pipeline that cross-compiles fluidsynth for iOS so we can provide those binaries during releases. The pipeline should work in a similar way as the Android pipeline. The pipeline itself should run on macOS (I think), because that's where [they have already installed all the iOS related SDKs](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#installed-sdks).

The compilation itself however differs from "usual" compilations, because (from my understanding) there will be only a single fat binary in the end, containing machine code for all architectures (armv7, arm64, anything else??). While CMake [seems to support this way of compilation](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#cross-compiling-for-ios-tvos-or-watchos) pretty well, it is a complete mystery to me how this works for AutoTools based projects (most of fluidsynth depending libs are AutoTools based).

So if anybody is able to shed some light or even propose a pipeline this would be much appreciated. Otherwise, I'll have to go through all the tutorials myself, and I can't tell if or when this will happen.
